ilastik/applets/edgeTraining/opEdgeTraining.py

```diff
@@ -387,37 +387,53 @@ class OpPredictEdgeProbabilities(Operator):
     TrainRandomForest = InputSlot(value=False)
     EdgeClassifier = InputSlot()
     EdgeFeaturesDataFrame = InputSlot()
+    # historically named Edge"Probabilities" because multicut used predictions from
+    # random forest as edge weights. With neural networks, edge probabilities are often already good
+    # enough to act as weights.
     EdgeProbabilities = OutputSlot()  # A 1D array of probabilities, in same order as EdgeFeaturesDataFrame
 
     def setupOutputs(self):
         self.EdgeProbabilities.meta.shape = (1,)
         self.EdgeProbabilities.meta.dtype = object
 
-    def execute(self, slot, subindex, roi, result):
-        pd.set_option("display.expand_frame_repr", False)
-        if self.TrainRandomForest.value:
-            edge_features_df = self.EdgeFeaturesDataFrame.value
-            classifier = self.EdgeClassifier.value
+    def _edge_weights_from_random_forest_predictions(self):
+        edge_features_df = self.EdgeFeaturesDataFrame.value
+        classifier = self.EdgeClassifier.value
 
-            # Classifier can be None if no labels have been selected
-            if classifier is None or len(classifier.known_classes) < 2:
-                result[0] = np.zeros((len(edge_features_df),), dtype=np.float32)
-                return
+        # Classifier can be None if no labels have been selected
+        if len(getattr(classifier, "known_classes", [])) == 0:
+            return np.zeros(len(edge_features_df), dtype=np.float32)
+
+        known_classes = classifier.known_classes
+        assert set(known_classes).issubset([1, 2])
+        if len(known_classes) == 1:
+            logger.info("Returning edge probabilities for only annotated class...")
+            if known_classes[0] == 1:
+                return np.zeros(len(edge_features_df), dtype=np.float32)
+            else:
+                return np.ones(len(edge_features_df), dtype=np.float32)
 
-            logger.info("Predicting edge probabilities...")
-            feature_matrix = edge_features_df.iloc[:, 2:].values  # Discard [sp1, sp2]
-            assert feature_matrix.dtype == np.float32, "Unexpected feature dtype: {}".format(feature_matrix.dtype)
-            probabilities = classifier.predict_probabilities(feature_matrix)[:, 1]
+        logger.info("Predicting edge probabilities...")
+        feature_matrix = edge_features_df.iloc[:, 2:].values  # Discard [sp1, sp2]
```

Comment:
Is it possible to select the required columns for the feature matrix by their names?

`feature_matrix = edge_features_df[:, ["col1", "col2"]].values`

Reply:
we do the names we want to exclude, the ones we want to include are not known in this operator (=all)

Reply:
Then we could discard them with `edge_features_df.drop(columns=("sp1", "sp2")).values`.